WordPress/Classes/ViewRelated/Plugins/PluginListViewController.swift

```diff
@@ -0,0 +1,79 @@
+import UIKit
+import WordPressKit
+
+class PluginListViewController: UITableViewController, ImmuTablePresenter {
+    fileprivate lazy var handler: ImmuTableViewHandler = {
+        return ImmuTableViewHandler(takeOver: self)
+    }()
+    fileprivate var viewModel: PluginListViewModel = .loading {
+        didSet {
+            handler.viewModel = viewModel.tableViewModelWithPresenter(self)
+            updateNoResults()
+        }
+    }
+
+    fileprivate let noResultsView = WPNoResultsView()
+
+    func updateNoResults() {
+        if let noResultsViewModel = viewModel.noResultsViewModel {
+            showNoResults(noResultsViewModel)
+        } else {
+            hideNoResults()
+        }
+    }
+
+    func showNoResults(_ viewModel: WPNoResultsView.Model) {
+        noResultsView.bindViewModel(viewModel)
+        if noResultsView.isDescendant(of: tableView) {
+            noResultsView.centerInSuperview()
+        } else {
+            tableView.addSubview(withFadeAnimation: noResultsView)
+        }
+    }
+
+    func hideNoResults() {
+        noResultsView.removeFromSuperview()
+    }
+
+    convenience init?(blog: Blog) {
+        precondition(blog.dotComID != nil)
+        guard let api = blog.wordPressComRestApi(),
+            let service = PluginServiceRemote(wordPressComRestApi: api) else {
+            return nil
+        }
+
+        self.init(siteID: Int(blog.dotComID!), service: service)
+    }
+
+    let siteID: Int
+    let service: PluginServiceRemote
```

Comment:
Personally I would add a new line after this line and I would move the convenience init to after the init below this comment, but that may be me just being particular about ordering things.